ui/store/actions.ts

```diff
@@ -454,7 +455,10 @@ export function addNewAccount(): ThunkAction<
 > {
   log.debug(`background.addNewAccount`);
   return async (dispatch, getState) => {
-    const oldAccounts = getInternalAccounts(getState());
+    const oldAccounts = getInternalAccounts(getState()).filter(
```

Comment:
I wonder if we could have something like `getInternalAccountsForKeyring(getState(), KeyringTypes.hd)`?

Reply:
Maybe you can even make it more generic to filter by different parameters, think that a lot of different accounts are going to be handled by the `eth-snap-keyring`

Reply:
This is, of course, out of scope for this PR

Reply:
Yes, that makes sense. I'll create this in another pr

Reply:
Sounds good!